validations/src/validations.rs

```diff
@@ -244,23 +247,50 @@ pub fn validate_rad_request(rad_request: &RADRequest) -> Result<(), failure::Err
     if retrieval_paths.is_empty() {
         return Err(DataRequestError::NoRetrievalSources.into());
     }
+    let mut is_rng = false;
     for path in retrieval_paths {
+        if path.kind == RADType::Rng {
+            if path.url.is_empty() || path.script.is_empty() {
+                is_rng = true;
+                break;
+            } else {
+                return Err(DataRequestError::InvalidRngRequest.into());
+            }
+        }
+
         // If the sources are empty the data request is set as invalid
         if path.url.is_empty() {
             return Err(DataRequestError::NoRetrievalSources.into());
         }
         unpack_radon_script(path.script.as_slice())?;
     }
 
+    if is_rng && retrieval_paths.len() > 1 {
+        return Err(DataRequestError::InvalidRngRequest.into());
+    }
```

Comment:
As per my general comment above, this goes against the three points.

Reply:
This only check that if it is RNG request, it should not have more sources